plugins/webhooks/index.js

```diff
@@ -2,10 +2,48 @@
 
 const boom = require('boom');
 const joi = require('joi');
+const winston = require('winston');
 const workflowParser = require('screwdriver-workflow-parser');
 
 const WAIT_FOR_CHANGEDFILES = 1.8;
 
+/**
+ * Update admins array
+ * @param  {UserFactory}    userFactory     UserFactory object
+ * @param  {String}         username        Username of user
+ * @param  {String}         scmContext      Scm which pipeline's repository exists in
+ * @param  {Pipeline}       pipeline        Pipeline object
+ * @return {Promise}                        Updates the pipeline admins and throws an error if not an admin
+ */
+async function updateAdmins(userFactory, username, scmContext, pipeline) {
+    try {
+        const user = await userFactory.get({ username, scmContext });
+        const userPermissions = await user.getPermissions(pipeline.scmUri);
+        const newAdmins = pipeline.admins;
+
+        // Delete user from admin list if bad permissions
+        if (!userPermissions.push) {
+            delete newAdmins[username];
+            // This is needed to make admins dirty and update db
+            pipeline.admins = newAdmins;
```

Comment:
`pipeline.admins = newAdmins;` and `return pipeline.update();` are duplicated with the below `if` statement. We can aggregate these.